frontend/src/components/MapView/Layers/LayerDropdown.tsx

```diff
@@ -95,12 +95,20 @@ function LayerDropdown({
       // 3. get rid of layers within the categories which don't match the given type
       .map(category => ({
         ...category,
-        layers: category.layers.filter(layer =>
-          layer.type === 'wms'
-            ? layer.type === type &&
-              [undefined, 'polygon'].includes(layer.geometry)
-            : layer.type === type,
-        ),
+        layers: category.layers.filter(layer => layer.type === type),
+      }))
+      // 4. additional filter for WMS layers
+      .map(category => ({
```

Comment:
I would suggest creating a function `filterLayersForAnalysis` that explicitly details the steps we use to filter out layers.

This way you can call it in the `.map` above and don't need to loop through the list one more time with a no-op for most categories / layers